tests/ert_tests/ensemble_evaluator/test_sync_ws_duplexer.py

```diff
@@ -44,18 +45,54 @@ def _start_ws(host: str, port: int, handler, ssl=None, sock=None):
     t.join()
 
 
-def test_immediate_stop():
-    duplexer = SyncWebsocketDuplexer("ws://localhost", "", None, None)
-    duplexer.stop()
+def test_immediate_stop(unused_tcp_port, ws):
+    # if the duplexer is immediately stopped, it should be well behaved and close
+    # the connection normally
+    closed = False
 
+    async def handler(websocket, path):
+        try:
+            await websocket.recv()
+        except ConnectionClosedOK:
+            nonlocal closed
+            closed = True
 
-def test_failed_connection():
-    with patch("ert.ensemble_evaluator.wait_for_evaluator") as w:
-        w.side_effect = OSError("expected oserror")
+    ws("localhost", unused_tcp_port, handler)
+    with ExitStack() as stack:
         duplexer = SyncWebsocketDuplexer(
-            "ws://localhost:0", "http://localhost:0", None, None
+            f"ws://localhost:{unused_tcp_port}",
+            f"ws://localhost:{unused_tcp_port}",
+            None,
+            None,
         )
-        with pytest.raises(OSError):
+        stack.callback(duplexer.stop)  # cleanup in case of failure
+
+        duplexer.stop()
+
+        assert closed
```

Comment:
I am bit rusty on `ExitStack`, but is the callback not always called when the `ExitStack` context is finished? In that case, would the second `duplexer.stop()` not be superfluous, if in addition the `assert closed` is taken out of the context manager?

Reply:
You are right! Thanks. Good call.